From 429464b51270f4f8068fe987b37aef9b1dab1a87 Mon Sep 17 00:00:00 2001
From: Peter Yin <peter.yin@quantatw.com>
Date: Wed, 29 Oct 2025 15:30:49 +0800
Subject: [PATCH 3/3] configuration: harma: set battery polling rate to 86400s

Set the battery polling rate to 86400 seconds (1 day) to reduce
unnecessary power drain from frequent polling.

Link to the https://gerrit.openbmc.org/c/openbmc/dbus-sensors/+/83271
implementation.

Change-Id: Iabec7f5d23d23c2b23c292dd20055355f9ec8d2b
Signed-off-by: Peter Yin <peter.yin@quantatw.com>
---
 configurations/meta/harma/harma_mb.json             | 1 +
 configurations/meta/harma/harma_mb_vr_infineon.json | 1 +
 schemas/legacy.json                                 | 4 ++++
 3 files changed, 6 insertions(+)

diff --git a/configurations/meta/harma/harma_mb.json b/configurations/meta/harma/harma_mb.json
index 2ea7dc4..5f9c0a7 100644
--- a/configurations/meta/harma/harma_mb.json
+++ b/configurations/meta/harma/harma_mb.json
@@ -121,6 +121,7 @@
                 ],
                 "Type": "ADC128D818",
                 "in0_Name": "MB_P3V_BAT_VOLT_V",
+                "in0_PollRate": 86400,
                 "in0_Scale": 333.333333,
                 "in1_Name": "MB_PVDD11_S3_VOLT_V",
                 "in1_PowerState": "On",
diff --git a/configurations/meta/harma/harma_mb_vr_infineon.json b/configurations/meta/harma/harma_mb_vr_infineon.json
index 3b28327..55f1f01 100644
--- a/configurations/meta/harma/harma_mb_vr_infineon.json
+++ b/configurations/meta/harma/harma_mb_vr_infineon.json
@@ -121,6 +121,7 @@
                 ],
                 "Type": "ADC128D818",
                 "in0_Name": "MB_P3V_BAT_VOLT_V",
+                "in0_PollRate": 86400,
                 "in0_Scale": 333.333333,
                 "in1_Name": "MB_PVDD11_S3_VOLT_V",
                 "in1_PowerState": "On",
diff --git a/schemas/legacy.json b/schemas/legacy.json
index 03846b1..194f324 100644
--- a/schemas/legacy.json
+++ b/schemas/legacy.json
@@ -817,6 +817,9 @@
                 "(curr|power|iout)(1[0-7]|[1-9])_Scale$": {
                     "$ref": "#/$defs/Types/ScaleValue"
                 },
+                "in(1[0-7]|[0-9])_PollRate$": {
+                    "$ref": "#/$defs/Types/PollRate"
+                },
                 "in(1[0-7]|[0-9])_PowerState$": {
                     "$ref": "#/$defs/Types/PowerState"
                 },
@@ -1376,6 +1379,7 @@
                 "type": "number"
             },
             "PollRate": {
+                "description": "Interval in seconds between consecutive polls",
                 "type": "number",
                 "exclusiveMinimum": 0
             }
-- 
2.47.3

